solr/core/src/java/org/apache/solr/packagemanager/RepositoryManager.java

```diff
@@ -0,0 +1,328 @@
+    }
+    if (version == null || "latest".equals(version)) {
+      return getLastPackageRelease(pkg);
+    }
+    for (SolrPackageRelease release : pkg.versions) {
+      if (PackageUtils.compareVersions(version, release.version) == 0) {
+        return release;
+      }
+    }
+    throw new SolrException(ErrorCode.BAD_REQUEST, "Package " + packageName + ":" + version + " does not exist in any repository.");
+  }
+
+  public SolrPackageRelease getLastPackageRelease(String packageName) {
+    SolrPackage pkg = getPackagesMap().get(packageName);
+    if (pkg == null) {
+      return null;
+    }
+    return getLastPackageRelease(pkg);
+  }
+
+  private SolrPackageRelease getLastPackageRelease(SolrPackage pkg) {
+    SolrPackageRelease latest = null;
+    for (SolrPackageRelease release: pkg.versions) {
+      if (latest == null) {
+        latest = release;
+      } else {
+        if (PackageUtils.compareVersions(latest.version, release.version) < 0) {
+          latest = release;
+        }
+      }
+    }
+    return latest;
+  }
+
+  /**
+   * Is there a version of the package available in the repositories that is more
+   * latest than our latest installed version of the package?
+   */
+  public boolean hasPackageUpdate(String packageName) {
+    SolrPackage pkg = getPackagesMap().get(packageName);
+    if (pkg == null) {
+      return false;
+    }
+    String installedVersion = packageManager.getPackageInstance(packageName, null).getVersion();
+    SolrPackageRelease last = getLastPackageRelease(packageName);
+    return last != null && PackageUtils.compareVersions(last.version, installedVersion) > 0;
+  }
+
+  /**
+   * Print list of available packages
+   */
+  public void listAvailable() throws SolrException {
+    PackageUtils.printGreen("Available packages:\n-----");
+    for (SolrPackage pkg: getPackages()) {
+      PackageUtils.printGreen(pkg.name + " \t\t"+pkg.description);
+      for (SolrPackageRelease version: pkg.versions) {
+        PackageUtils.printGreen("\tVersion: "+version.version);
+      }
+    }
+  }
+
+  /**
+   * Install a version of the package. Also, run verify commands in case some
+   * collection was using "$LATEST" version of this package and got auto-updated.
+   */
+  public void install(String packageName, String version) throws SolrException {
+    String latestVersion = getLastPackageRelease(packageName).version;
+
+    Map<String, String> collectionsDeployedIn = packageManager.getDeployedCollections(packageName);
+    List<String> peggedToLatest = collectionsDeployedIn.keySet().stream().
+        filter(collection -> collectionsDeployedIn.get(collection).equals("$LATEST")).collect(Collectors.toList());
+    if (!peggedToLatest.isEmpty()) {
+      PackageUtils.printGreen("Collections that will be affected (since they are configured to use $LATEST): "+peggedToLatest);
+    }
+
+    if (version == null || version.equals("latest")) {
```

Comment:
If user wants to install mypackage or mypackage:latest, it has the same effect here. It will find the most recent package in repository and install it.